smac/utils/sklearn_wrapper.py

```diff
@@ -0,0 +1,392 @@
+              - None, to use the default 3-fold cross validation,
+              - integer, to specify the number of folds in a `(Stratified)KFold`,
+              - An object to be used as a cross-validation generator.
+              - An iterable yielding train, test splits.
+
+            For integer/None inputs, if the estimator is a classifier and ``y`` is
+            either binary or multiclass, :class:`StratifiedKFold` is used. In all
+            other cases, :class:`KFold` is used.
+
+            Refer :ref:`User Guide <cross_validation>` for the various
+            cross-validation strategies that can be used here.
+
+        refit : boolean, default=True
+            Refit the best estimator with the entire dataset.
+            If "False", it is impossible to make predictions using
+            this RandomizedSearchCV instance after fitting.
+
+        verbose : integer
+            Controls the verbosity: the higher, the more messages.
+
+        random_state : int or RandomState
+            Pseudo random number generator state used for random uniform sampling
+            from lists of possible values instead of scipy.stats distributions.
+
+        error_score : 'raise' (default) or numeric
+            Value to assign to the score if an error occurs in estimator fitting.
+            If set to 'raise', the error is raised. If a numeric value is given,
+            FitFailedWarning is raised. This parameter does not affect the refit
+            step, which will always raise the error.
+
+        return_train_score : boolean, default=True
+            If ``'False'``, the ``cv_results_`` attribute will not include training
+            scores.
+
+        intensification_percentage : float, default=None
+            SMAC parameter, the fraction of time to be used on intensification
+            (versus choice of next Configurations). In case of None, the SMAC
+            default will be used.
+            """
+
+        # Current implementation does not like distributions yet.
+        for param, values in param_distributions.items():
+            if not isinstance(values, list):
+                raise ValueError('Not implemented (yet): Wrapper does not work with distributions yet. Please use lists. ')
+        # For now, this is not supported
+        scoring = None
+
+        self.random_state = random_state
+        self.param_distributions = param_distributions
+        self.n_iter = n_iter
+        self.intensification_percentage = intensification_percentage
+
+        # sets important hyperparameters of search module
+        super(ModelBasedOptimization, self).__init__(
+              estimator=estimator, scoring=scoring, fit_params=fit_params,
+              n_jobs=1, iid=iid, refit=refit, cv=cv, verbose=verbose,
+              pre_dispatch=False, error_score=error_score,
+              return_train_score=return_train_score)
+
+    def fit(self, X: Union[List[List], np.array], y: Union[List, np.array, None] = None,
```

Comment:
In scikit-learn, X could be many more things. Like a list of strings or a dataframe.

Reply:
And it should actually be `np.ndarray`, `np.array` is a function.